File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.cpp
class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n=nums.size();

        int minIdx = 0, maxIdx=0;

        for(int i=0; i<n; i++){
            if(nums[i] < nums[minIdx]) minIdx= i;
            if(nums[i] > nums[maxIdx]) maxIdx= i;
        }

        int left = min(maxIdx, minIdx);
        int right = max(maxIdx, minIdx);

        return min({right + 1, n - left, left+1 + n - right});
    }
};

File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        minIdx = nums.index(min(nums))
        maxIdx = nums.index(max(nums))
        n = len(nums)
        left = min(minIdx, maxIdx)
        right = max(minIdx, maxIdx)
        return min(right+1, n-left, left+1+n-right)
        